### ISSUE TITLE
Bug: [no-misused-spread] False negative for spread in JSX

### ISSUE THREAD

**Issue by OliverJAsh** [typescript-eslint/typescript-eslint#10648]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gDwFzwAoBKeAXgD5CBveAMySX0xgEsUBzeAXyIG4AofgHoh8QKDk8ENgAO4DCGCSYMJDAA08dkiwhlq-gRoA6E9m59BI%2BIBlySTLkKlK9Zu3wUqEPwA8wFgDd4KhMjbC54IXJeIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9GABM4MAF8QSoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```tsx
declare const x: () => ({ foo: string });

// ✅ expected error, got error
({ ...x });

// ❌ expected error, got none
<div {...x} />;
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-misused-spread": 2
  }
}
```

### Expected Result

See above.

### Actual Result

See above.

### Additional Info

It looks like we're matching against `SpreadElement` but not `JSXSpreadAttribute`.
